torch/utils/file_baton.py

```diff
@@ -32,16 +32,24 @@ def try_acquire(self):
         except FileExistsError:
             return False
 
-    def wait(self):
+    def wait(self, show_lock_path=False):
         """
         Periodically sleeps for a certain amount until the baton is released.
 
         The amount of time slept depends on the ``wait_seconds`` parameter
         passed to the constructor.
         """
+        displayed_lock_path = False
+
+        tik = time.time()
         while os.path.exists(self.lock_file_path):
             time.sleep(self.wait_seconds)
 
+            if show_lock_path:
+                if time.time() - tik > 2000 and not displayed_lock_path:
+                    print(f"Waiting lock file: {self.lock_file_path}")
```

Comment:
Use warnings.warn instead?

Reply:
Would the message here still be something like, "uh oh we've waited for a while, maybe something gone wrong - here's the lock file path btw, maybe you want to delete it"